services/src/main/java/org/keycloak/authorization/authorization/AuthorizationTokenService.java

```diff
@@ -430,7 +460,9 @@ private EvaluationContext createEvaluationContext(KeycloakAuthorizationRequest r
         BiFunction<KeycloakAuthorizationRequest, AuthorizationProvider, EvaluationContext> evaluationContextProvider = SUPPORTED_CLAIM_TOKEN_FORMATS.get(claimTokenFormat);
 
         if (evaluationContextProvider == null) {
-            throw new CorsErrorResponseException(request.getCors(), OAuthErrorException.INVALID_REQUEST, "Claim token format [" + claimTokenFormat + "] not supported", Status.BAD_REQUEST);
+            CorsErrorResponseException unsupportedClaimTokenFormatException = new CorsErrorResponseException(request.getCors(), OAuthErrorException.INVALID_REQUEST, "Claim token format [" + claimTokenFormat + "] not supported", Status.BAD_REQUEST);
+            fireErrorEvent(request.getEvent(), Errors.UNSUPPORTED_CLAIM_TOKEN_FORMAT, unsupportedClaimTokenFormatException);
```

Comment:
I would just use Errors.INVALID_REQUEST. The details should be enough to understand what happened.

Reply:
agreed